revolutionpi/revolution_pi.go

```diff
@@ -154,5 +154,35 @@ func (b *revolutionPiBoard) DoCommand(ctx context.Context,
 		resp["getStatus"] = "status ok"
 	}
 
+	pinMessage, exists := req["readAddress"]
+	if exists {
+		pinName, ok := pinMessage.(string)
+		if !ok {
+			return nil, fmt.Errorf("error performing readAddress: expected string got %v", pinMessage)
+		}
+		pin := SPIVariable{strVarName: char32(pinName)}
+		err := b.controlChip.mapNameToAddress(&pin)
+		if err != nil {
+			return nil, err
+		}
+		b.controlChip.logger.Debugf("pin: %#v", pin)
+		switch pin.i16uLength {
+		case 1:
+			value, err := b.controlChip.getBitValue(int64(pin.i16uAddress), pin.i8uBit)
+			if err != nil {
+				return nil, err
+			}
+			resp[pinName] = value
+		default:
+			value := make([]byte, pin.i16uLength/8)
+			n, err := b.controlChip.fileHandle.ReadAt(value, int64(pin.i16uAddress))
+			if err != nil {
+				return nil, err
+			}
+			b.controlChip.logger.Debugf("Read %#d bytes", n)
+			resp[pinName] = readFromBuffer(value, n)
+		}
+	}
```

Comment:
if they try to do a command that doesn't exist should we return an error?

Reply:
I can set it up that way! Probably a good idea to do this since we would be documenting the commands